File: 347-top-k-frequent-elements/347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # Create empty dictionary to store seen numbers
        # Loop through nums, for element we:
        # Check if number is in dict, if it is, increment the count
        # If it isn't add it to the dict
        # Return sorted(dict.keys())[:k]
        
        seen = {}
        for num in nums:
            if num in seen:
                seen[num] += 1
            else:
                seen[num] = 1
        
        return sorted(seen.keys(), key=lambda x: seen[x], reverse=True)[:k]

File: 347-top-k-frequent-elements/347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        // Use hashmap to keep track of seen values
        // Loop through nums
        // For each num, increment the count by one in hashmap
        // return k most frequent elements
        
        unordered_map<int, int> seen {};       
        for (auto& num : nums) {
            seen[num] += 1;
        }
        
        priority_queue<pair<int,int>> pQueue {};
        for (auto [key, value] : seen) {
            pQueue.push({value, key});
        }
        
        vector<int> ans {};
        while(k-- && !pQueue.empty()){
            ans.push_back(pQueue.top().second);
            pQueue.pop();
        }
        return ans;
    }
};